org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/correction/UnresolvedTypesQuickFixTest.java

```diff
@@ -1576,4 +1575,31 @@ public void testMultipleAddAllMissingImports() throws Exception {
 		assertEquals(1, addAllMissingImportsActions.size());
 	}
 
+	@Test
+	public void testIgnoreTypeFilter() throws Exception {
+		IPackageFragment pack1 = fSourceFolder.createPackageFragment("test1", false, null);
+		StringBuilder buf = new StringBuilder();
+		buf.append("package test1;\n");
+		buf.append("import java.util.ArrayList;\n");
+		buf.append("public class E {\n");
+		buf.append("    void foo() {\n");
+		buf.append("        List v= new ArrayList();\n");
+		buf.append("    }\n");
+		buf.append("}\n");
+		ICompilationUnit cu = pack1.createCompilationUnit("E.java", buf.toString(), false, null);
+
+		buf = new StringBuilder();
+		buf.append("package test1;\n");
+		buf.append("import java.util.ArrayList;\n");
+		buf.append("import java.util.List;\n");
+		buf.append("public class E {\n");
+		buf.append("    void foo() {\n");
+		buf.append("        List v= new ArrayList();\n");
+		buf.append("    }\n");
+		buf.append("}\n");
```

Comment:
How about we started using text blocks, rather than String concatenation via StringBuilders?

Reply:
I like the idea!